src/toast/Toast.tsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import {Icon} from "../icon";
+import {classNames} from "../utils";
+import {ClassNames} from "../modals/ClassNames";
+
+export enum Direction {
+    TOP_RIGHT = "top-right",
+    BOTTOM_LEFT = "bottom-left",
+    BOTTOM_RIGHT = "bottom-right",
+    TOP_LEFT = "top-left",
+}
+
+export enum MessageType {
+    SUCCESS = "success",
+    WARNING = "warning",
+    DANGER = "danger",
+    INFO = "info",
+}
+
+const DEFAULT_HIDE_TIME = 3000;
+
+type ToastProps = {
+    showToast?: boolean;
+    duration?: number;
+    position?: Direction;
+    type?: MessageType;
+    icon?: string;
+    text: string;
+    onClose?: Function;
+    style?: any;
+    className?: string;
+};
+
+type ToastState = {
+    showToast: boolean;
+};
+
+export class Toast extends React.PureComponent<ToastProps> {
+    private divRef: HTMLDivElement | null = null;
+    // Remove the warning for setState on componentUnmount using isMounted
+    private _isMounted = false;
+
+    state: ToastState = {
+        showToast: this.props.showToast !== undefined ? this.props.showToast : false,
+    };
+
+    // Default value for props if no value is given
+    static defaultProps = {
+        showToast: false,
+        type: MessageType.SUCCESS,
+        position: Direction.TOP_RIGHT,
+        duration: DEFAULT_HIDE_TIME,
+    };
+
+    componentWillUpdate(nextProps: ToastProps, nextState: ToastState) {
+        if (this.showToast(nextProps, nextState)) {
+            this.createElement();
+
+            if (nextProps.showToast === true && nextProps.showToast !== nextState.showToast) {
+                //set the timer for the toast
+                this.setTimer(nextProps.duration!);
+            }
+        } else {
+            this.cleanup();
+        }
+    }
+
+    componentWillMount() {
+        this.createElement();
+    }
+
+    createElement() {
+        if (this.divRef === null) {
+            const el = document.createElement("div");
+            document.body.appendChild(el);
+            this.divRef = el;
+            document.body.classList.add(ClassNames.NO_SCROLLING);
+        }
+    }
+
+    componentDidMount() {
+        this._isMounted = true;
+    }
+
+    setTimer(hideTime: number) {
+        setTimeout(() => {
+            this.close();
+        }, hideTime);
+    }
+
+    cleanup() {
+        if (this.divRef !== null) {
+            document.body.removeChild(this.divRef);
+            this.divRef = null;
+            document.body.classList.remove(ClassNames.NO_SCROLLING);
+        }
+    }
+
+    componentWillUnmount() {
+        this.close();
```

Comment:
Why do we call this.close() here?

Reply:
I was facing an issue where if the toast is visible and I navigate to another page and then come back to the toast page then the previous toast is still visible as it was not closed earlier (because component unmounted while navigating to the other page). To fix this I have added this.close() in componentWillUnmount() to close the toast if we are navigating to another page while the toast is visible.

Reply:
Removed this.close() from componentWillUnmount.

Reply:
Yeah, I think this is the right decision.  It seems a little difficult to cover all use-cases in Storybook environment.  Hopefully, in a true we UI the Toast will be higher up in the component chain a can still display/hide across multiple pages.